firebase-installations/src/main/java/com/google/firebase/installations/AwaitListener.java

```diff
@@ -0,0 +1,38 @@
+// Copyright 2019 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.firebase.installations;
+
+import androidx.annotation.NonNull;
+import com.google.android.gms.tasks.OnCompleteListener;
+import com.google.android.gms.tasks.Task;
+import java.util.concurrent.CountDownLatch;
+import java.util.concurrent.TimeUnit;
+
+final class AwaitListener implements OnCompleteListener<Void> {
+  private final CountDownLatch mLatch = new CountDownLatch(1);
+
+  public void onSuccess() {
```

Comment:
Its used. If the status is REGISTERED, instead of waiting for 10s, I am using this method to update the awaitlistener.

FirebaseInstalllations.java line 261